---
title: "Data Preprocessing"
output: 
  html_document:
    toc: true
    toc_float: true
  
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In this part, I will walk you through details of our data and introduce some data proprecessing procedures.  I hope you could get a clear understanding of our data and are able to generate you own training dataset after this session.  

Basicaly, we will use rat in vivo liver diffrential gene expression data as training data and use the correspoding pathology terms as labels. For your quickly understanding , imagine our training data as a matrix whose rows represent samples and coloumns represent features. In more details, each sample(row) represents diffrential gene expression levels in liver for a  rat at certain time point and cetain drug dose level. And each feature(column) represents expression level of a certain gene across different samples. Each sample has a pathological record for each pathology from TG-GATEs.

---

## Diffrential gene expression data
 
### Gene expression data 
Before we go further, let's first introduce what is gene expression data. Gene is the basic physical and functional unit of heredity among all the species. Slight difference between genes decide individual's unique physical features. Gene controls the traits or phenotypes of species by transcripting and tranlating to RNAs and proteins. The expression level of transcription and translation  can be dedected by [microarrays](https://en.wikipedia.org/wiki/DNA_microarray) or next generation sequencing techniques(NGS), which is called gene expression data.

The gene expression levels in rat in vivo liver in [TG-GATEs][Link to TG-GATEs paper] are measured by microarrays. So the results are stored in .CEL file format and all the CEL data pass the qualitity control  [QC](https://www.ncbi.nlm.nih.gov/pubmed/15364885)

Then let's get a intuitive understanding of gene expression data from rat in vivo liver that administered acarbose in single-dose experiment(See below what is [single-dose experiment](#single_dose_experiment) in TG-GATEs). In the folder of "TG-GATEs/rat/in_vivo/Liver/Single/acarbose.Rat.in_vivo.Liver.Single",
there is a folder called "celfiles" containing CEL files for each specific drug administered experiment. Another Attribute.csv file in this folder includes metadata for each CEL file in "celfiles" directory. 

<center><img src="images/acarbose_folder.png" width="900px"></center>


For example, the second row in Attribute.csv file have a attribute "BARCODE" with value "3017401004"

<center><img src="images/acarbose_attributes.png" width="900px"></center>


This means this row is meatadata for "003017401004.CEL" in "celfiles" folder. 

<center><img src="images/acarbose_celfiles.png" width="900px"></center>


The attibutes "ORGAN", "COMPOMD_NAME","SPECIES","TEST_TYPE","SIN_REP_TYPE","SACRI_PERIOD" and "DOSE_level " in second row of attributes file tell us "003017401004.CEL" containing the gene expression data of liver for rat administerted acarbose, which was measured at 3 hour with dose level "control". Notice for each experimental condition,there are several samples(i.e. Several CELfiles) to [increase statistical power](https://www.illumina.com/Documents/products/technotes/technote_power_replicates.pdf)

Then we load all the CEL data in this folder and preprocess them with R packages ["affy"](https://www.bioconductor.org/packages/release/bioc/html/affy.html). This is necessary and reasons for this  are explained in [This paper](https://www.ncbi.nlm.nih.gov/pubmed/14960456 )

* Read all the CEL files from acarbose administered experiment by function "ReadAffy" to "AffyBatch" object

```{r}

library(affy)
cell_dir="../TGGATEs_tutorial_secrete/acarbose.Rat.in_vivo.Liver.Single"
fns=list.celfiles(path=paste(cell_dir,"celfiles",sep = "/"),full.names=TRUE)
data=ReadAffy(filenames = fns)

```



* Then we can use function "rma" in "affy" package to preprocess our data by following steps:
  1. Background correction 
  2. Normaliation 
  3. Computation of expression data 
Function "rma"" will transfer our "AffyBatch" object into "ExpressionSet" object. ["ExpressionSet" object](https://www.bioconductor.org/packages/3.7/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf) is a class frequently used  in genomics to manage high-throughput assays and experimental meatadata. 


```{r}
eset=rma(data,normalize = TRUE)
head(exprs(eset))

```

"exprs" function returns expression data of our experiment. Columns refer to samples and rows refer to [probeset](https://www.ensembl.org/info/genome/microarray_probe_set_mapping.html) in microarray.

* From "Atrribute.tsv"" file, we can add "phenoData" to "ExpressionSet" to describe samples in the experiment

```{r}
pheno=read.table(paste(cell_dir,"/Attribute.tsv",sep = ""),sep = "\t",header = TRUE,fileEncoding="latin1")
pheno=pheno[which(pheno[,"BARCODE"]!="No ChipData"),]
rownames(pheno)=paste("00",sub('^0+',"",pheno[,"BARCODE"]),".CEL",sep = "") 
pheno=pheno[colnames(exprs(eset)),]
pData(eset) = pheno
head(pData(eset))
```


* Annotate probesets of microarray to gene names by [ribiosAnnotation](https://github.com/Accio/ribios/tree/master/ribiosAnnotation) to describe features in the experiment.
Since [each gene is usually represented by more than on probeset](https://www.biostars.org/p/54632/), we use [ribiosUtils](https://github.com/Accio/ribios/tree/master/ribiosUtils) to further preprocess data to get unique gene names for our data .

```{r}
library(ribiosAnnotation)
library(ribiosUtils)
geneName=annotateProbesets(featureNames(eset), orthologue = TRUE)
fData(eset) <- geneName
eset_rmNA=eset[!is.na(fData(eset)[,"GeneID"]),]
index_uniqueProbe=isMaxStatRow(exprs(eset_rmNA),keys = fData(eset_rmNA)[,"GeneSymbol"])
eset=eset_rmNA[index_uniqueProbe,]
```

* After we get expression data, we remove gene with expression level less than 6 as these data are  unreliable
```{r}
eset <- eset[apply(exprs(eset), 1, max, na.rm=TRUE)>6,]
```




### Diffrential gene expression data 
But here we use diffrential gene expression data other than gene expression data mainly because of [batch effect](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3880143/) as  TG-GATEs  is a long period(10 year) project conducted  by various  institutes and companies. But you  could also try  gene expression data since you can get more samples in this case.

Diffrential gene expression data means quantitative changes of gene expression level between different experimental groups.  Here we calculate diffrential gene expression for each experimental condition  compared with control experimental condition.

The method we used to caculate diffrential gene expression data is called limma

```{r}
time_points=unique(pData(eset)$SACRI_PERIOD)
limmaTG<-function(eset_timei_contrast,filename){
   f=factor(eset_timei_contrast$DOSE_LEVEL)
   # get dose information
   uni_levels=unique(f)
   level=uni_levels[-(which(uni_levels=="Control"))]
   # get csww information,  csww contains compound type, species, in vitro/vivo and organism information 
   csww=tail(unlist(strsplit(filename,"/")),n=1)
   #get time information
   timepoint=unique(pData(eset_timei_contrast)$SACRI_PERIOD)
   # get sample name 
   sample_name=paste(csww,timepoint,level,sep = "/")
   design=model.matrix(~+f)
   fit=lmFit(eset_timei_contrast,design)
   fit=eBayes(fit)
   nenv=new.env()
   #add logFC data as experssion data
   fit_topTable <- topTable(fit,coef = 2,number = nrow(fit$genes))
   fit_exprs=matrix(fit_topTable$logFC)
   #rownames(fit_exprs)=rownames(fit$genes)  ## errors here ,fuck 
   rownames(fit_exprs)=fit_topTable$ProbeID  ## when using some function, you'd better understand its 
                                             ## function exactly 
   colnames(fit_exprs)=sample_name
   #plot(fit_exprs,fit_topTable$logFC)
   assign("exprs",fit_exprs,nenv)
  
   fit_eset=ExpressionSet(assayData = nenv)
   fit_pheno=data.frame(time=timepoint,dose=level,csww=csww) 
   rownames(fit_pheno)=colnames(fit_exprs)
   pData(fit_eset)=fit_pheno
   
   fData(fit_eset)=fit$genes
   return(fit_eset)
   
 }
 esets=list()
 for (i in 1:length(time_points)) {
   eset_timei=eset[,which(pData(eset)$SACRI_PERIOD==time_points[i])]
   dose_levels=unique(pData(eset_timei)$DOSE_LEVEL)
   dose_levels=dose_levels[-which(dose_levels=="Control")]
   eset_timei_contrasts=list()
   for (ii in 1:length(dose_levels)) {
     eset_timei_contrasts[[ii]]=eset_timei[,c(which(eset_timei$DOSE_LEVEL=="Control"),which(eset_timei$DOSE_LEVEL==dose_levels[ii]))]
   }
   esets[[i]]=lapply(eset_timei_contrasts, limmaTG,filename=cell_dir)
 }
 #combine all expressin data of this certain compound into one
 compound_esets=list()
 index=1
 for (i in 1:length(esets)) {
   for (j in 1:length(esets[[i]])) {
     #index=(i-1)*length(esets[[i]])+j
     compound_esets[[index]]=esets[[i]][[j]]
     index=index+1
   }
 }
 fenv=new.env()
 #combine logFC
 #notice toptable in diffrent in different limma analysis!!!!!!!!!!!!!!!!!!!!!!!!!
 compound_common_features=rownames(compound_esets[[1]])
 for (i  in 2:length(compound_esets) ) {
   compound_common_features=intersect(compound_common_features,rownames(compound_esets[[i]]))
 }
 
 for (i in 1:length(compound_esets)) {
   compound_esets[[i]]=compound_esets[[i]][compound_common_features,]
 }
 
 compound_exprs=sapply(compound_esets,exprs)
 rownames(compound_exprs)=rownames(compound_esets[[1]])
 colnames(compound_exprs)=sapply(compound_esets, function(x){colnames(exprs(x))})
 assign("exprs",compound_exprs,fenv)
 
 
 compound_final_eset=ExpressionSet(assayData = fenv)
 compound_pheno=lapply(compound_esets, pData)
 compound_pheno=do.call("rbind",compound_pheno)
 pData(compound_final_eset)=compound_pheno
 
 fData(compound_final_eset)=fData(compound_esets[[1]])
 saveRDS(compound_final_eset,file = paste(cell_dir,".rds",sep = ""))
```




## TG-GATEs


### <a id="single_dose_experiment"></a> Experimental design

## <a id="data_preprocessing_reasons"></a> Data preprocessing 

why data preprocessing


## Data combination






[Link to TG-GATEs paper]: https://www.ncbi.nlm.nih.gov/pubmed/25313160

---























